kornia/augmentation/_2d/intensity/random_rain.py

```diff
@@ -37,45 +39,46 @@ def __init__(
+        self, image: Tensor, params: dict[str, Tensor], flags: dict[str, Any], transform: Tensor | None = None
     ) -> Tensor:
+        height_of_drop = params['drop_height_factor']
+        width_of_drop = params['drop_width_factor']
+
+        # Check array
         KORNIA_CHECK(image.shape[1] == 3 or image.shape[1] == 1, "Number of color channels should be 1 or 3.")
         KORNIA_CHECK(len(image.shape) in (3, 4), "Wrong input dimension.")
-
         for i in range(image.shape[0]):
-            number_of_drops: int = int(params['number_of_drops_factor'][i])
-            height_of_drop: int = int(params['drop_height_factor'][i])
-            width_of_drop: int = int(params['drop_width_factor'][i])
-            KORNIA_CHECK(height_of_drop <= image[i].shape[1], "Height of drop should be less than image height")
+            KORNIA_CHECK(int(height_of_drop[i]) <= image[i].shape[1], "Height of drop should be less than image height")
 
-            KORNIA_CHECK(width_of_drop <= image[i].shape[2], "Width of drop should be less than image width")
-            KORNIA_CHECK(height_of_drop > 0, "Height should be bigger than 0")
+            KORNIA_CHECK(int(width_of_drop[i]) <= image[i].shape[2], "Width of drop should be less than image width")
+            KORNIA_CHECK(int(height_of_drop[i]) > 0, "Height should be bigger than 0")
             # Generate start coordinates for each drop
 
-            random_y_coords = torch.randint(low=0, high=image[i].shape[1] - height_of_drop, size=[1, number_of_drops])
-            if width_of_drop < 0:
-                random_x_coords = torch.randint(
-                    low=-width_of_drop - 1, high=image[i].shape[2], size=[1, number_of_drops]
-                )
+        for i in range(image.shape[0]):
+            coordinates_of_drops = params['coordinates_factor'][i]
+            height_of_drop = params['drop_height_factor'][i]
+            width_of_drop = params['drop_width_factor'][i]
+
+            random_y_coords = coordinates_of_drops[:, 0] * (image.shape[2] - height_of_drop - 1)
+            if width_of_drop > 0:
+                random_x_coords = coordinates_of_drops[:, 1] * (image.shape[3] - width_of_drop - 1)
             else:
-                random_x_coords = torch.randint(
-                    low=0, high=image[i].shape[2] - width_of_drop, size=[1, number_of_drops]
-                )
-            coords = torch.cat([random_y_coords, random_x_coords], dim=0).to(image.device)
+                random_x_coords = coordinates_of_drops[:, 1] * (image.shape[3] + width_of_drop - 1) - width_of_drop
+
+            coords = torch.cat([random_y_coords[None], random_x_coords[None]], dim=0).to(image.device, dtype=torch.long)
 
             # Generate how our drop will look like into the image
             size_of_line = max(height_of_drop, abs(width_of_drop))
             x = torch.linspace(start=0, end=height_of_drop, steps=size_of_line, dtype=torch.int32).to(image.device)
```

Comment:
```suggestion
            x = torch.linspace(start=0, end=height_of_drop, steps=size_of_line, device=image.device, dtype=torch.int32)
```
This allocates directly. Is the dtype correct, or should be long?

Reply:
x couldn't be bigger than the shape of the image, so int32 will be enough